rust/operator-binary/src/config.rs

```diff
@@ -0,0 +1,184 @@
+            ),
+            user_registration_role = authentication_method
+                .ldap_extras
+                .as_ref()
+                .map(|extra| &extra.user_registration_role)
+                .unwrap_or(&stackable_superset_crd::default_user_registration_role()),
+            roles_sync_at_login = to_python_bool(
+                authentication_method
+                    .ldap_extras
+                    .as_ref()
+                    .map(|extra| &extra.sync_roles_at)
+                    .unwrap_or(&stackable_superset_crd::default_sync_roles_at())
+                == &LdapRolesSyncMoment::Login
+            ),
+        }
+        .as_str(),
+    );
+
+    // Possible TLS options, see https://github.com/dpgaspar/Flask-AppBuilder/blob/f6f66fc1bcc0163a213e4a2e6f960e91082d201f/flask_appbuilder/security/manager.py#L243-L250
+    match &ldap.tls {
+        None => config.push_str("AUTH_LDAP_USE_TLS = False\n"),
+        Some(tls) => match &tls.verification {
+            TlsVerification::None {} => {
+                config.push_str(indoc! {r#"
+                    AUTH_LDAP_USE_TLS = False # This setting is for STARTTLS which we dont support at the moment, we use dedicated Tls.
+                    AUTH_LDAP_ALLOW_SELF_SIGNED = True
+                "#});
+            }
+            TlsVerification::Server(server_verification) => {
+                append_server_ca_cert(
+                    config,
+                    authentication_class_name,
+                    &server_verification.ca_cert,
+                );
+            }
+            TlsVerification::Mutual(mutual_verification) => {
+                append_server_ca_cert(
+                    config,
+                    authentication_class_name,
+                    &CaCert::SecretClass(mutual_verification.cert_secret_class.to_string()),
+                );
+                config.push_str(formatdoc! {r#"
+                    AUTH_LDAP_TLS_CERTFILE = "/certificates/{authentication_class_name}-tls-certificate/tls.crt"
+                    AUTH_LDAP_TLS_KEYFILE = "/certificates/{authentication_class_name}-tls-certificate/tls.key"
+                "#}
+                .as_str());
+            }
+        },
+    }
+
+    if ldap.bind_credentials.is_some() {
+        config.push_str(
+        formatdoc! {r#"
+                AUTH_LDAP_BIND_USER = open('/secrets/{authentication_class_name}-bind-credentials/user').read()
+                AUTH_LDAP_BIND_PASSWORD = open('/secrets/{authentication_class_name}-bind-credentials/password').read()
+            "#}
+            .as_str()
+        );
+    }
+}
+
+fn append_server_ca_cert(
+    config: &mut String,
+    authentication_class_name: &str,
+    server_ca_cert: &CaCert,
+) {
+    config.push_str(
+        indoc! {r#"
+            AUTH_LDAP_USE_TLS = False # This setting is for STARTTLS which we dont support at the moment, we use dedicated Tls.
```

Comment:
This doesn't really have much to do with the CA...

Reply:
I actually remove the STARTTLS setting completely